app/views/statuses/_group.html.erb

```diff
@@ -0,0 +1,13 @@
+<div class="status-group">
+  <%= render 'statuses/status', status: group %>
+
+  <div class="status-items">
+    <% group.statuses.each do |status| %>
```

Comment:
> why StatusGroup have all those methods.

So that we can feed it to the regular `_status` partial. Commits with a single status go directly to the `_status` partial. It's a way to use polymorphism to avoid too many `if / else` in views.